pymc3/distributions/simulator.py

```diff
@@ -51,16 +101,44 @@ def random(self, point=None, size=None):
         -------
         array
         """
-
-        raise NotImplementedError("Not implemented yet")
+        params = draw_values([*self.params], point=point, size=size)
+        if size is None:
+            return self.function(*params)
```

Comment:
if `self.function` is a theano function it will return a tensor - maybe we can push the `draw_values` call to the end and return that?

Reply:
self.function is intended to be a Python function not a Theano one.